python/ray/serve/controller.py

```diff
@@ -519,6 +538,28 @@ async def __init__(self,
 
         asyncio.get_event_loop().create_task(self.run_control_loop())
 
+    async def wait_for_event(self, uuid: ResultId) -> FutureResult:
+        future = self.inflight_results[uuid]
+        await future.wait()
+        # NOTE(ilr) We have to delete this *after* we return to avoid a
+        # situation where the controller crashes *after* deleting and before
+        # returning to the API (and user).
+        asyncio.get_event_loop().call_soon_threadsafe(
+            lambda: self.inflight_results.pop(uuid))
+        return future.future_result
+
+    def _create_and_set_result(self, result: bool,
+                               goal_state: Dict[str, any]) -> ResultId:
+        fut = WrappedResult()
+        fut.future_result = FutureResult(result, goal_state)
```

Comment:
This will override the class member for everyone. https://docs.python.org/3/tutorial/classes.html#class-and-instance-variables

Instead of inheriting from asyncio.Event, you can just do `asyncio.get_event_loop().create_future()` and later call `future.set_result()` or `await future`.

Reply:
Oh! Ofc!